Solutions/Threat Intelligence/Analytic Rules/DomainEntity_PaloAlto.yaml

```diff
@@ -23,79 +23,74 @@ triggerThreshold: 0
+      // Filter to pick up only IOC's that contain the entities we want (in this case, DomainName)
+      | where isnotempty(DomainName)
+      | extend TI_DomainEntity = DomainName;
   Domain_Indicators
-  // using innerunique to keep perf fast and result set low, we only need one match to indicate potential malicious activity that needs to be investigated
-  | join kind=innerunique (
-  CommonSecurityLog
-  | extend IngestionTime = ingestion_time()
-  | where IngestionTime > ago(dt_lookBack)
-  | where DeviceVendor =~ 'Palo Alto Networks'
-  | where DeviceEventClassID =~ 'url'
-  //Uncomment the line below to only alert on allowed connections
-  //| where DeviceAction !~ "block-url"
-  //Extract domain from RequestURL, if not present extarct it from AdditionalExtentions
-  | extend PA_Url = column_ifexists("RequestURL", "None")
-  | extend PA_Url = iif(isempty(PA_Url) and AdditionalExtensions !startswith "PanOS", extract("([^\"]+)", 1, tolower(AdditionalExtensions)), trim('"', PA_Url))
-  | extend PA_Url = iif(PA_Url !startswith "http://" and ApplicationProtocol !~ "ssl", strcat('http://', PA_Url), iif(PA_Url !startswith "https://" and ApplicationProtocol =~ "ssl", strcat('https://', PA_Url), PA_Url))
-  | extend Domain = trim(@"""",tostring(parse_url(PA_Url).Host))
-  | where isnotempty(Domain)
-  | extend Domain = tolower(Domain)
-  | extend parts = split(Domain, '.')
-  //Split out the TLD for the purpose of checking if we have any TI indicators with this TLD to match on
-  | extend tld = parts[(array_length(parts)-1)]
-  //Validate parsed domain by checking TLD against TLDs from threat feed and drop domains where there is no chance of a match
-  | where tld in~ (list_tlds)
-  | extend CommonSecurityLog_TimeGenerated = TimeGenerated
-  ) on $left.TI_DomainEntity==$right.Domain
-  | where CommonSecurityLog_TimeGenerated < ExpirationDateTime
-  | summarize CommonSecurityLog_TimeGenerated = arg_max(CommonSecurityLog_TimeGenerated, *) by IndicatorId, Domain, IoC
-  | project CommonSecurityLog_TimeGenerated, Description, ActivityGroupNames, PA_Url, Domain, IndicatorId, ThreatType, ExpirationDateTime, ConfidenceScore,
-    DeviceAction, DestinationIP, DestinationPort, DeviceName, SourceIP, SourcePort, ApplicationProtocol, RequestMethod, Type, IoC, TI_DomainEntity
-  | extend timestamp = CommonSecurityLog_TimeGenerated
+      // Join with CommonSecurityLog to find potential malicious activity
+      | join kind=innerunique (
+          CommonSecurityLog
+          | extend IngestionTime = ingestion_time()
+          | where IngestionTime > ago(dt_lookBack)
+          | where DeviceVendor =~ 'Palo Alto Networks'
+          | where DeviceEventClassID =~ 'url'
+          // Uncomment the line below to only alert on allowed connections
+          // | where DeviceAction !~ "block-url"
+          // Extract domain from RequestURL, if not present, extract it from AdditionalExtensions
+          | extend PA_Url = column_ifexists("RequestURL", "None")
```

Comment:
Validated and i can see empty cells in RequestURL, So _ifexists condition is necessary

Reply:
Column_ifexists is required when the field in not present in the schema at all. If you want to put value of "None" in case there is no value in RequestURL column, please use iff function or coalesce function like as follows:
| extend PAURL = coalesce(RequestURL, "None")